src/requestHandlers/panelSet.ts

```diff
@@ -163,6 +169,133 @@ const getAllTrunkSets = async(request: Request, res: Response) : Promise<Respons
     */
 };
 
+interface PanelSetNode {
+    panel_set_id: number,
+    parent_panel_set_id: number | null,
+    author_id: string,
+    level: number,
+    created_at: DataTypes.DateDataType,
+    updated_at: DataTypes.DateDataType,
+    path: string
+}
+
+interface PanelSetFrontEnd {
+    panel_set_id: number,
+    parent_panel_set_id: number | null,
+    author_id: string,
+    created_at: DataTypes.DateDataType,
+    updated_at: DataTypes.DateDataType,
+    childrenIds: number[]
+}
+
+const getTree = async(request: Request, res: Response) : Promise<Response> => {
+    const panel_set_id = request.params.id;
+    const response = await _getTreeController(sequelize)(Number(panel_set_id)) as PanelSetNode[];
+
+    //! calling sanitizeResponse more than once throws an error unless it's a return statement
+    //! probably not the best way to check if an error is thrown
+    if(!Array.isArray(response)) {
+        return sanitizeResponse(response, res);
+    }
+    response.sort((a : PanelSetNode, b : PanelSetNode) => a.level - b.level)
+    const panel_sets = [] as PanelSetFrontEnd[];
+    for(const panel_set of response) {
+        const children = response.filter(p => p.parent_panel_set_id == panel_set.panel_set_id);
+        panel_sets.push({
+            panel_set_id: panel_set.panel_set_id,
+            author_id: panel_set.author_id,
+            created_at: panel_set.created_at,
+            updated_at: panel_set.updated_at,
+            childrenIds: children.map(p => p.panel_set_id)
+        } as PanelSetFrontEnd)
+    }
+    return sanitizeResponse(panel_sets, res, 'Custom 404 error');;
+}
+
+const _getTreeController = (sequelize: Sequelize) => async(id: number) => {
+    try {
+        const response = await _getPanelSetByIDController(sequelize)(id);
+        //if an error or contains not a panelSet return
+        if(response instanceof Error) {
+            return response;
+        }
+        
+        //? This isn't the best way to check if it's a panel set, but unsure of another way
+        if(!(response as any).author_id) {
```

Comment:
I am unsure of how to do that. 
```ts
response instanceof IPanelSet
```
gives several syntax errors, main one being `'IPanelSet' only refers to a type, but is being used as a value here.`

Reply:
`response as IPanelSet` maybe?

Reply:
that doesn't return the boolean, but rather the variable casted. It may work for when `_getPanelSetByIDController` returns null, but that's not really what I'm looking for and is pretty much the same as what I currently have.

Reply:
Yeah you are right, probably fine to leave it as is then.